### ISSUE TITLE
Which APK to download to install SPMC?

### ISSUE THREAD

**Issue by ATMRLZ** [koying/SPMC#932]
I'm new to this and I already got my Android Emulator ready because I don't see Windows download available anymore. Below are the choices of APK's to download. What is the difference and which do I need??? Thank you so much for the help.

SPMC-16.7.0-spmc-1e54f91-aarch64.apk
SPMC-16.7.0-spmc-1e54f91-aarch64_launcher.apk
SPMC-16.7.0-spmc-1e54f91-armeabi-v7a.apk
SPMC-16.7.0-spmc-1e54f91-armeabi-v7a_launcher.apk
SPMC-16.7.0-spmc-1e54f91-x86.apk
SPMC-16.7.0-spmc-1e54f91-x86_launcher.apk

**Comment by sammarbella** [comment]
Depends on the CPU/SoC your device use:

- aarch64 is for 64 bit CPU armv8 SoC
- v7a is for 32 bit CPU armv7 SoC (can also be used in armv8 SoC)
- is for x86 CPU based devices (PC CPUs)

"launcher" builds can be used to set the app as device launcher

**Comment by alslan** [comment]
I wanted to install on my Android 6.01 TV Box Zidoo X9s with Realtek SoC RTD1295 the 64-bit version: SPMC-16.7.0-spmc-1e54f91-aarch64.apk.
The installation failed. The SoC is 64-bit, because support ARM64

**Comment by ATMRLZ** [comment]
I have a Sony Vaio 64bit operating system so I would install the aarch64 . As for the launcher, I'm not understanding. So I would install the launcher if I wanted SPMC to be a stand-a-lone like Kodi instead of it being a app inside lets say a emulator? Sorry for being new at this lol. It's so different than Kodi with the install.

**Comment by sammarbella** [comment]
I guess you have a Vaio laptop running Windows.

SPMC has .exe Windows builds...

Download a Windows build for it.

http://download.semperpax.com/spmc/win32/

Use the last modified .exe

**Comment by sammarbella** [comment]
@alslan 

Google is your friend:

http://forum.zidoo.tv/index.php?threads/kodi-17-0-seems-working-fine.4222/

Read starting post 10 there.

Use 32 bit builds for armv7...

**Comment by ATMRLZ** [comment]
Oh okay, yeah that is what I was trying to find. Youtube videos on it showed the Windows build but on the site I couldn't find it until you linked that. Thank you.

Also, one video I viewed, the guy said the download somehow overwrites into the Kodi folder because the name, is that true? I want to use SPMC but I don't want to lose my Kodi. Thank you again for your help.

**Comment by sammarbella** [comment]
Kodi and SPMC have separate folders, you can have both installed without problem.

Just don't run them at the same time. lol

**Comment by ATMRLZ** [comment]
LMAO.  That is good to know haha. I really appreciate you. I will try the link you gave me for the Windows build and see. Updates follow.

**Comment by ATMRLZ** [comment]
Okay so I downloaded it. Told me the error about the missing libexpat.dll so I fixed that from another thread on here. SPMC comes up and the header scrolls with news but there is NO menu to click. I hit enter and just nothing. I can't go into any menus because there isn't any...

**Comment by ATMRLZ** [comment]
NO MENUS. It loads up just fine but there isn't ANY menus to click. Something is wrong and IDK what to do. I downloaded the latest .exe from the link you gave and everything was a breeze but this program isn't doing anything inside

**Comment by alslan** [comment]
I think SPMC works only on Android 5.x +
Not on Windows and not in emulators, because the graphics acceleration is missing.
There are exe installation files, but they have never worked properly with me under Windows 7 x64.

**Comment by sammarbella** [comment]
@ATMRLZ 

Just installed in my Windows 10, works fine.

@alslan 

http://spmc.semperpax.com/

> Jarvis
As far as I’m concerned, Nvidia Shield has basically cornered the market, so is my main development device. More generally, Android TV devices, amlogic devices and Amazon FireTV might be getting some love ;)
Important: No more Rockchip specific support as of this version

> Note1: As of 16.5.0, Android lollipop (API 21) is required

**Comment by ATMRLZ** [comment]
I have Windows 7 and all I see is the space man background and the long stripe bar where the menu is supposed to be but nothing is there to click, So weird...

**Comment by alslan** [comment]
@ sammarbella

I use FTMC, is for Rockchip SoC, is better for me. But same Problem with Bluetooth Fire TV Stick 1. Generation Remote, press once is registered as two times. With SPMC 13.4.0, which I actually use constantly, i have no such problem. i changed keymap.xml accordingly.

**Comment by alslan** [comment]
I have SPMCSetup-20161227-972797d-16.5.x.exe installed and which works on Windows 7 x64 (graphics card is an Nvidia GT 630M 1 GB with driver from 06.2016(It is not supported h.265 hardwareside). I could play a h.265 2160p video file on my LG 43UH603v.
Unfortunately, frameswitching does not work. So when playing there is judder.

**Comment by ATMRLZ** [comment]
@alslan 

It worked on my Emulator, well the menus and downloading apps. But when I play a movie, it's just a black screen and just sound. 

As for my Windows that's where it's absolutely no menus to click so I have to Uninstall it. 

Any idea how to fix the black screen I get from the Google play app SPMC? Like I said, I can hear the movie or show but can't see it. I'm giving up on the SPMC for Windows.

**Comment by alslan** [comment]
Try using FTMC. I have unfortunately no Sony Vaio laptop and can not help you.

[http://freaktab.com/forum/tv-player-support/general-tv-player-dicussions/xbmc-talk/575588-ftmc](url)
[If you use the link and it comes to the error message, the link does not exist, in the new window the link copy/paste and confirm there.]
There are two versions, 16.1 and 16.2 test both. These are based on Kodi 16.2 or 16.1.:
FTMC-16.2.29-armeabi-v7a.apk
FTMC-16.1-armeabi-v7a-Build-26.apk

Because of Google play, I had a Chinese TV box and that could only download certain programs. I sold it. With Zidoo X9s, I can install SPMC from Google Play store. But costs about $120, the Zidoo X9s I mean.

**Comment by sammarbella** [comment]
@alslan 

I'm sorry, i can't help you with that.

I considered to buy a cheap solution like Amazon Fire TV or even a 100$ chinese box but i finally decided to pay premium and avoid problems: i bought a 200$ Shield TV.

No regrets so far, it's the best media player since it was launched 2 years ago.

**Comment by alslan** [comment]
I had ever bought Fire TV 1 generation, for 49 €. It was very good, I have stopped through the router the updates (OpenWRT) t and quite expensive again.
It has not supported h.265, which has disturbed me. When playing h.265 1080p content has always jerks.

**Comment by ATMRLZ** [comment]
So I have been posting on other pages and people are telling me the same issue they have that I am having on this Windows 32 link. It's something with the software to where it's not giving me the font or anything to click into and use. Nothing to do with my windows 7 or laptop I have. I'm just lost. 

Now what is this FTMC?

**Comment by LilSnoop40** [comment]
what version for first generation firetv:

16.7.0-spmc-1e54f91-aarch64.apk
16.7.0-spmc-1e54f91-aarch64_launcher.apk
16.7.0-spmc-1e54f91-armeabi-v7a.apk
16.7.0-spmc-1e54f91-armeabi-v7a_launcher.apk
16.7.0-spmc-1e54f91-x86.apk
16.7.0-spmc-1e54f91-x86_launcher.apk

Thanks!

**Comment by sammarbella** [comment]
https://developer.amazon.com/public/solutions/devices/fire-tv/docs/device-and-platform-specifications

Fire TV get 1 has a 32 bit SoC, use armv7 builds.

Fire TV gen 2 has a 64 bit SoC, use arch64 builds (armv7 will also works).

x86 builds are only for PC CPUs.it's not strictly like that: Nexus Player use an Intel Atom CPU but  you can still call it a "PC CPU"

**Comment by ATMRLZ** [comment]
Still a no go for me with my windows 7 Sony Vaio lol. I just can't get it. I might try like other forks but Idk.

**Comment by sammarbella** [comment]
That's very strange.It must be your Windows 7/ hardware. :s 

If you are curious enough you can spend a few minutes and try to run it in a VM Windows 10 in VirtualBox:

[https://betanews.com/2015/07/29/how-to-install-windows-10-on-oracle-virtualbox-no-windows-key-required/ ](url)

or VMWare Player (free):

[https://www.tenforums.com/tutorials/8135-install-windows-10-virtual-machine-vmware-player.html](url)

**Comment by unlocKing** [comment]
SPMC for Windows will use your existing Kodi folder for addons and usedata, so if you have had any previous addons/skins settings you will see that:

More info here:
https://github.com/koying/SPMC/issues/676

**Comment by ATMRLZ** [comment]
That's what I thought but it didn't overwrite my Kodi. As was said before you can use both on your PC with no problems.

**Comment by unlocKing** [comment]
Think you missed the point, it will not overwrite your Kodi installation folders.

Your data will be not be separated and will shared between SPMC and Kodi, if that has no consequence to you then. 

When you go back and forth between Kodi and SPMC you will start noticing it.

**Comment by Gertbucket** [comment]
@sammarbella Been using the armv7 builds on my AFTV2. Has anyone noticed any advantage in using the 64bit build?

**Comment by ATMRLZ** [comment]
Yeah I haven't noticed anything because the f'n app doesn't work lol. But now you got me wondering if my Kodi isn't going to work right now haha.
